tools/shared/src/main/scala/org/scalajs/core/tools/linker/analyzer/SymbolRequirement.scala

```diff
@@ -0,0 +1,102 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package org.scalajs.core.tools.linker.analyzer
+
+sealed trait SymbolRequirement {
+  final def ++(that: SymbolRequirement): SymbolRequirement =
+    SymbolRequirement.multipleInternal(List(this, that))
+}
+
+object SymbolRequirement {
+  import Nodes._
+
+  def factory(originatingComponent: String): Factory =
+    new Factory(originatingComponent)
+
+  final class Factory private[SymbolRequirement] (origin: String) {
+    def accessModule(moduleName: String): SymbolRequirement =
+      AccessModule(origin, moduleName)
+
+    def callOnModule(moduleName: String, methodName: String): SymbolRequirement =
+      multiple(accessModule(moduleName), callMethod(moduleName, methodName))
+
+    def instantiateClass(className: String,
+        constructor: String): SymbolRequirement = {
+      InstantiateClass(origin, className, constructor)
+    }
+
+    def instantiateClass(className: String,
+        constructors: Traversable[String]): SymbolRequirement = {
```

Comment:
Well you could just `require(constructors.nonEmpty)` ...

But then also, it's not really a big deal if it *is* empty, although it might be confusing.
Alternatively, you could just get rid of this method altogether. It is only used for the constructors of `RuntimeLong`.